{
  "uid" : "d825904bc7cf5af9",
  "name" : "test_revoke_approve",
  "fullName" : "tests.brownie.token.tests.test_approve#test_revoke_approve",
  "historyId" : "493792eebd5a4fea1ce221a16db97b7d",
  "time" : {
    "start" : 1642073999609,
    "stop" : 1642073999609,
    "duration" : 0
  },
  "status" : "broken",
  "statusMessage" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f11670e35e0>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
  "statusTrace" : "brownie_project = <Project 'TokenProject'>\naccounts = <brownie.network.account.Accounts object at 0x7f405705bdc0>\n\n    @pytest.fixture(scope=\"module\")\n    def token(brownie_project, accounts):\n>       return brownie_project.Token.deploy(\"Test Token\", \"TST\", 18, 1e21, {'from': accounts[0]})\n\ntests/brownie/token/tests/conftest.py:49: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/contract.py:600: in __call__\n    return tx[\"from\"].deploy(\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py:508: in deploy\n    receipt, exc = self._make_transaction(\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py:725: in _make_transaction\n    raise VirtualMachineError(e) from None\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = VirtualMachineError(ValueError(\"Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries ex...ion refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\"))\nexc = \"Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by N...ction refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\"\n\n    def __init__(self, exc: ValueError) -> None:\n        try:\n            exc = exc.args[0]\n        except Exception:\n            pass\n    \n        if isinstance(exc, dict) and \"message\" in exc:\n            if \"data\" not in exc:\n                raise ValueError(exc[\"message\"]) from None\n    \n            self.message: str = exc[\"message\"].rstrip(\".\")\n    \n            if isinstance(exc[\"data\"], str):\n                # handle parity exceptions - this logic probably is not perfect\n                if \"0x08c379a0\" in exc[\"data\"]:\n                    revert_type, err_msg = [i.strip() for i in exc[\"data\"].split(\"0x08c379a0\", 1)]\n                    err_msg = eth_abi.decode_abi([\"string\"], HexBytes(err_msg))\n                    err_msg = f\"{revert_type} '{err_msg}'\"\n                elif exc[\"data\"].endswith(\"0x\"):\n                    err_msg = exc[\"data\"][:-2].strip()\n                else:\n                    err_msg = exc[\"data\"]\n                raise ValueError(f\"{self.message}: {err_msg}\") from None\n    \n            try:\n                txid, data = next((k, v) for k, v in exc[\"data\"].items() if k.startswith(\"0x\"))\n            except StopIteration:\n                raise ValueError(exc[\"message\"]) from None\n    \n            self.txid: str = txid\n            self.source: str = \"\"\n            self.revert_type: str = data[\"error\"]\n            self.pc: Optional[str] = data.get(\"program_counter\")\n            if self.pc and self.revert_type == \"revert\":\n                self.pc -= 1\n    \n            self.revert_msg: Optional[str] = data.get(\"reason\")\n            self.dev_revert_msg = brownie.project.build._get_dev_revert(self.pc)\n            if self.revert_msg is None and self.revert_type in (\"revert\", \"invalid opcode\"):\n                self.revert_msg = self.dev_revert_msg\n            elif self.revert_msg == \"Failed assertion\":\n                self.revert_msg = self.dev_revert_msg or self.revert_msg\n    \n        else:\n>           raise ValueError(str(exc)) from None\nE           ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f11670e35e0>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\n\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/exceptions.py:121: ValueError",
  "flaky" : false,
  "newFailed" : false,
  "newBroken" : true,
  "newPassed" : false,
  "retriesCount" : 0,
  "retriesStatusChange" : false,
  "beforeStages" : [ {
    "name" : "configure_neon_network",
    "time" : {
      "start" : 1642073985866,
      "stop" : 1642073985866,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "brownie_project",
    "time" : {
      "start" : 1642073987038,
      "stop" : 1642073998129,
      "duration" : 11091
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "accounts",
    "time" : {
      "start" : 1642073998589,
      "stop" : 1642073998857,
      "duration" : 268
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "setup_env",
    "time" : {
      "start" : 1642073998589,
      "stop" : 1642073998589,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "token",
    "time" : {
      "start" : 1642073998857,
      "stop" : 1642073999148,
      "duration" : 291
    },
    "status" : "broken",
    "statusMessage" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f11670e35e0>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\n",
    "statusTrace" : "  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/pluggy/_callers.py\", line 39, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/_pytest/fixtures.py\", line 1126, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/_pytest/fixtures.py\", line 932, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/neon-compatibility/neon-compatibility/tests/brownie/token/tests/conftest.py\", line 49, in token\n    return brownie_project.Token.deploy(\"Test Token\", \"TST\", 18, 1e21, {'from': accounts[0]})\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/contract.py\", line 600, in __call__\n    return tx[\"from\"].deploy(\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py\", line 508, in deploy\n    receipt, exc = self._make_transaction(\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py\", line 725, in _make_transaction\n    raise VirtualMachineError(e) from None\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/exceptions.py\", line 121, in __init__\n    raise ValueError(str(exc)) from None\n",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  } ],
  "afterStages" : [ {
    "name" : "accounts::0",
    "time" : {
      "start" : 1642074028414,
      "stop" : 1642074028414,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "feature",
    "value" : "Brownie"
  }, {
    "name" : "parentSuite",
    "value" : "tests.brownie.token.tests"
  }, {
    "name" : "suite",
    "value" : "test_approve"
  }, {
    "name" : "host",
    "value" : "fv-az128-463"
  }, {
    "name" : "thread",
    "value" : "3767-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.brownie.token.tests.test_approve"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 24,
        "skipped" : 0,
        "passed" : 71,
        "unknown" : 0,
        "total" : 95
      },
      "items" : [ {
        "uid" : "1e4028b87353de7e",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/483//#testresult/1e4028b87353de7e",
        "status" : "passed",
        "time" : {
          "start" : 1642059397169,
          "stop" : 1642059400854,
          "duration" : 3685
        }
      }, {
        "uid" : "a039b35725a74d67",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/482//#testresult/a039b35725a74d67",
        "status" : "passed",
        "time" : {
          "start" : 1642043718343,
          "stop" : 1642043720834,
          "duration" : 2491
        }
      }, {
        "uid" : "4fa28370ec13d68e",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/481//#testresult/4fa28370ec13d68e",
        "status" : "passed",
        "time" : {
          "start" : 1642034803527,
          "stop" : 1642034807329,
          "duration" : 3802
        }
      }, {
        "uid" : "73736b3c2bfa0dc",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/479//#testresult/73736b3c2bfa0dc",
        "status" : "passed",
        "time" : {
          "start" : 1642032968342,
          "stop" : 1642032972413,
          "duration" : 4071
        }
      }, {
        "uid" : "f6000371c0046633",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/478//#testresult/f6000371c0046633",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.HTTPError: 502 Server Error: Bad Gateway for url: http://proxy.night.stand.neontest.xyz/solana",
        "time" : {
          "start" : 1642032071012,
          "stop" : 1642032071012,
          "duration" : 0
        }
      }, {
        "uid" : "427eaec5be112033",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/477//#testresult/427eaec5be112033",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642031186739,
          "stop" : 1642031186739,
          "duration" : 0
        }
      }, {
        "uid" : "1be18c8edcf45bb2",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/476//#testresult/1be18c8edcf45bb2",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642028348698,
          "stop" : 1642028348698,
          "duration" : 0
        }
      }, {
        "uid" : "4cd4427f00962abd",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/474//#testresult/4cd4427f00962abd",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642025670987,
          "stop" : 1642025670987,
          "duration" : 0
        }
      }, {
        "uid" : "8a0410592b14c907",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/472//#testresult/8a0410592b14c907",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642018235611,
          "stop" : 1642018235611,
          "duration" : 0
        }
      }, {
        "uid" : "7b7441450f44c717",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/471//#testresult/7b7441450f44c717",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642015602803,
          "stop" : 1642015602803,
          "duration" : 0
        }
      }, {
        "uid" : "7aaa940a2537fd7f",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/470//#testresult/7aaa940a2537fd7f",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642014228611,
          "stop" : 1642014228611,
          "duration" : 0
        }
      }, {
        "uid" : "cdf3b6a754ff331",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/469//#testresult/cdf3b6a754ff331",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642012910263,
          "stop" : 1642012910263,
          "duration" : 0
        }
      }, {
        "uid" : "d858b4d8227306ed",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/468//#testresult/d858b4d8227306ed",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642011420326,
          "stop" : 1642011420326,
          "duration" : 0
        }
      }, {
        "uid" : "15f60cf44d6201a5",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/466//#testresult/15f60cf44d6201a5",
        "status" : "passed",
        "time" : {
          "start" : 1641997947676,
          "stop" : 1641997951796,
          "duration" : 4120
        }
      }, {
        "uid" : "fedb58734f0648be",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/465//#testresult/fedb58734f0648be",
        "status" : "passed",
        "time" : {
          "start" : 1641993709922,
          "stop" : 1641993714002,
          "duration" : 4080
        }
      }, {
        "uid" : "15293e24db153a69",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/464//#testresult/15293e24db153a69",
        "status" : "passed",
        "time" : {
          "start" : 1641990970173,
          "stop" : 1641990973826,
          "duration" : 3653
        }
      }, {
        "uid" : "d64ccc5e7b80aab5",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/463//#testresult/d64ccc5e7b80aab5",
        "status" : "passed",
        "time" : {
          "start" : 1641986332723,
          "stop" : 1641986335554,
          "duration" : 2831
        }
      }, {
        "uid" : "45c95f3dd04a5e46",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/461//#testresult/45c95f3dd04a5e46",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f18961e3b80>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641982476962,
          "stop" : 1641982476962,
          "duration" : 0
        }
      }, {
        "uid" : "e13c82f7abf4e548",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/460//#testresult/e13c82f7abf4e548",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f189611ea30>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641980265827,
          "stop" : 1641980265827,
          "duration" : 0
        }
      }, {
        "uid" : "581c468df08e3c4",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/459//#testresult/581c468df08e3c4",
        "status" : "passed",
        "time" : {
          "start" : 1641975784808,
          "stop" : 1641975788496,
          "duration" : 3688
        }
      } ]
    },
    "tags" : [ ]
  },
  "source" : "d825904bc7cf5af9.json",
  "parameterValues" : [ ]
}